Gem/Code/Source/UserSettings/MultiplayerSampleUserSettings.cpp

```diff
@@ -55,159 +86,268 @@ namespace MultiplayerSample
+                entityContextId, &AzFramework::GameEntityContextRequestBus::Events::GetGameEntityContextId);
+
+            if (auto scene = AZ::RPI::Scene::GetSceneForEntityContextId(entityContextId); scene)
             {
-                AZ::Data::Instance<AZ::RPI::StreamingImagePool> pool = imageSystem->GetSystemStreamingPool();
-                pool->SetMipBias(textureQuality);
+                if (auto reflectionFeatureProcessor =
+                    scene->GetFeatureProcessor<AZ::Render::SpecularReflectionsFeatureProcessorInterface>(); reflectionFeatureProcessor)
+                {
+                    auto ssrOptions = reflectionFeatureProcessor->GetSSROptions();
+                    ssrOptions.m_enable = (reflectionType != SpecularReflections::None);
+                    ssrOptions.m_rayTracing = (reflectionType == SpecularReflections::ScreenSpaceAndRaytracing);
+                    reflectionFeatureProcessor->SetSSROptions(ssrOptions);
+                }
             }
+        }
+
+        m_registry->Set(ReflectionSettingKey.c_str(), aznumeric_cast<AZ::u64>(reflectionType));
+    }
+
+    Msaa MultiplayerSampleUserSettings::GetMsaa()
+    {
+        AZ::u64 msaa = static_cast<AZ::u64>(DefaultMsaa);
+        m_registry->Get(msaa, MsaaKey.c_str());
+
+        return static_cast<Msaa>(msaa);
+    }
 
-            registry->Set(m_textureQualityKey.c_str(), aznumeric_cast<AZ::s64>(textureQuality));
+    void MultiplayerSampleUserSettings::SetMsaaInRenderer(Msaa msaa)
+    {
+        if (auto rpiSystem = AZ::RPI::RPISystemInterface::Get(); rpiSystem)
+        {
+            auto multisampleState = rpiSystem->GetApplicationMultisampleState();
+            switch (msaa)
+            {
+            case Msaa::X1:
+                multisampleState.m_samples = 1;
+                break;
+            case Msaa::X2:
+                multisampleState.m_samples = 2;
+                break;
+            case Msaa::X4:
+                multisampleState.m_samples = 4;
+                break;
+            }
+            rpiSystem->SetApplicationMultisampleState(multisampleState);
         }
     }
 
-    bool MultiplayerSampleUserSettings::GetFullscreen()
+    void MultiplayerSampleUserSettings::SetMsaa(Msaa msaa)
     {
-        bool fullscreen = DefaultFullscreenMode;
+        // Currently MSAA settings don't get changed at runtime because they have the potential
+        // to cause a TDR graphics driver crash on at least Vulkan, maybe others.
+        // This might be the result of mixing PSOs between the msaa variant and the non-msaa variant in the same frame.
+        // Until the problem gets tracked down and resolved, we'll only set the MSAA setting in the renderer at boot time.
+        // If this ever gets fixed, the call to SetMsaaInRenderer() can get moved to here, and the extra flow handling to call
+        // it sooner can get removed.
+
+        m_registry->Set(MsaaKey.c_str(), aznumeric_cast<AZ::u64>(msaa));
```

Comment:
Switching between 2x to 4x should work fine. Only 1x to non-1x should require a restart. Although this is also fine for now.

Reply:
Yeah, I deliberately chose to make them all work the same way. Otherwise, you end up with a Settings toggle that only works immediately 2/3 of the time and requires a restart 1/3 of the time, which just looks broken.